odbc/src/api/data.rs

```diff
@@ -1398,6 +1399,88 @@ pub unsafe fn get_diag_rec(
     )
 }
 
+///
+/// get_stmt_diag_field copies a part of the given ODBC error's diagnostic information
+/// into the provided pointer.
+///
+/// # Safety
+/// This writes to a raw C-pointer
+///
+pub unsafe fn get_stmt_diag_field(
+    handle: &Statement,
+    diag_identifier: DiagType,
+    diag_info_ptr: Pointer,
+) -> SqlReturn {
+    match diag_identifier {
+        DiagType::RowCount => {
+            *(diag_info_ptr as *mut u64) = 0u64;
+            SqlReturn::SUCCESS
+        }
+        DiagType::RowNumber => {
+            *(diag_info_ptr as *mut u64) = handle.attributes.read().unwrap().row_number as u64;
```

Comment:
I think that if the value is 0, it means that the error is not associated with a specific row and you need to return SQL_NO_ROW_NUMBER.
Also, since I don't think that we are updating this value, I would consider returning SQL_ROW_NUMBER_UNKNOWN unconditionally and adding the work to update the attribute value and return the correct value to the JIRA for handling the missing diagnostic fields.

Reply:
I think `SQL_ROW_NUMBER_UNKOWN` would make sense here, will add.